test/stepper.js

```diff
@@ -62,6 +62,349 @@ exports["Stepper Firmware Requirement"] = {
+            supportedModes: [8],
+          },
+        ]
+      }),
+      debug: false,
+      repl: false
+    });
+
+    this.board2 = new Board({
+      io: new MockFirmata({
+        pins: [
+          {
+            supportedModes: [8],
+          },
+        ]
+      }),
+      debug: false,
+      repl: false
+    });
+
+    done();
+  },
+
+  tearDown: function(done) {
+    done();
+  },
+
+  singleBoard: function(test) {
+    test.expect(2);
+
+    test.doesNotThrow(function() {
+      for (var i = 0; i < this.maxSteppers; i++) {
+        new Stepper({
+          board: this.board1,
+          stepsPerRev: 200,
+          pins: [3, 4, 5, 6]
+        });
+      }
+    }.bind(this));
+
+    test.throws(function() {
+      new Stepper({
+        board: this.board1,
+        stepsPerRev: 200,
+        pins: [3, 4, 5, 6]
+      });
+    }.bind(this));
+
+    test.done();
+  },
+
+  multipleBoards: function(test) {
+    test.expect(3);
+
+    // should be able to add max on two boards
+    test.doesNotThrow(function() {
+      for (var i = 0; i < this.maxSteppers; i++) {
+        new Stepper({
+          board: this.board1,
+          stepsPerRev: 200,
+          pins: [2, 3, 4, 5]
+        });
+        new Stepper({
+          board: this.board2,
+          stepsPerRev: 200,
+          pins: [2, 3, 4, 5]
+        });
+      }
+    }.bind(this));
+
+    // but can't add any more to either board
+    test.throws(function() {
+      new Stepper({
+        board: this.board1,
+        stepsPerRev: 200,
+        pins: [2, 3, 4, 5]
+      });
+    }.bind(this));
+    test.throws(function() {
+      new Stepper({
+        board: this.board2,
+        stepsPerRev: 200,
+        pins: [2, 3, 4, 5]
+      });
+    }.bind(this));
+
+    test.done();
+  }
+};
+
+// TODO test that callback is called for .step()
+// mock firmata apparently never calls it, so we can't test currently
+// exports["Stepper - step callback"] = {};
+
+exports["Stepper - set direction required before step"] = {
+  setUp: function(done) {
+
+    this.board = new Board({
+      io: new MockFirmata({
+        pins: [
+          {
+            supportedModes: [8],
+          },
+        ]
+      }),
+      debug: false,
+      repl: false
+    });
+
+    this.stepper = new Stepper({
+      board: this.board,
+      type: five.Stepper.TYPE.DRIVER,
+      stepsPerRev: 200,
+      pins: [2, 3]
+    });
+    done();
+  },
+
+  tearDown: function(done) {
+    done();
+  },
+
+  directionSet: function(test) {
+    test.expect(1);
+
+    // If we haven't set a direction, we should get error
+    this.stepper.step(1, function(error) {
+      test.ok(!!error);
+    });
+
+    // after setting direction, we shouldn't
+    this.stepper.cw();
+    this.stepper.step(1, function(error) {
+      if (!!error) {
+        // Testing that we shouldn't get here.
+        // If we do, expected number of tests will fail
+        test.ok(false, "should not have gotten error after setting direction");
```

Comment:
Not sure about pattern elsewhere, but I like your suggestion. I arrived at this after testing several different ways and didn't notice the obvious (now that you point it out) simplification. I'll make this change, but we might want to handle it differently based on my previous comment regarding error codes.

What I'm specifically trying to test is that it throws a "must set direction" error if you don't call `.cw()` or `.ccw()` before `.step()` but doesn't throw that specific error after a direction is set.

I'd like this test to check for that exact error condition and let other tests handle other errors that might be thrown from `.step()`

thoughts?